src/webui/index.js

```diff
@@ -183,17 +187,34 @@ module.exports = async function () {
   })
 
   const launchWebUI = ctx.getFn('launchWebUI')
+  const splashScreenPromise = ctx.getProp('splashScreen')
+  if (store.get(CONFIG_KEY) && process.env.NODE_ENV !== 'test') {
+    // we're supposed to show the window on startup, display the splash screen
+    (await splashScreenPromise).show()
+  } else {
+    // we don't need the splash screen, ignore it.
+    (await splashScreenPromise).destroy()
+  }
 
   return /** @type {Promise<void>} */(new Promise(resolve => {
-    window.once('ready-to-show', async () => {
-      logger.info('[web ui] window ready')
-
-      if (store.get(CONFIG_KEY)) {
-        await launchWebUI('/')
-      }
-
-      resolve()
-    })
+    if (store.get(CONFIG_KEY)) {
+      logger.info('[web ui] waiting for ipfsd to start')
+      window.once('ready-to-show', async () => {
+        logger.info('[web ui] window ready')
+
+        const interval = setInterval(async () => {
+          if (![null, STATUS.STARTING_STARTED].includes(ipfsdStatus)) {
```

Comment:
We show the webui window when ipfsdStatus is not set to `null` (initial value) or `STATUS.STARTING_STARTED`, the first value it's set to when it's booting up. This means the webui window will show whether ipfsd successfully connects to the node or not.